src/pages/index.js

```diff
@@ -22,128 +23,261 @@ import {
+    return api
+      .removeLike(cardId)
+      .then(() => false)
+      .catch((err) => {
+        console.error("Like Button Error:", err);
+        return true;
+      });
+  }
+}
+
+//// Button Event Listeners ////
+// avatar image click
+profileAvatarWrapper.addEventListener("click", openEditAvatarModal);
+
 // profile edit button click
 profileEditButton.addEventListener("click", openEditProfileModal);
 
 // add card button click
 addCardButton.addEventListener("click", openAddCardModal);
 
-//// Forms ////
-const profileForm = document.forms["edit-profile-form"];
-const addCardForm = document.forms["add-card-form"];
+//// Form Submit Handlers ////
+// Avatar form submission handler
+function handleAvatarFormSubmit(data) {
+  profileAvatar.src = data.link;
 
-//// Form Input Fields ////
-const profileNameInput = profileForm.querySelector(".modal__input_type_name");
-const profileDescriptionInput = profileForm.querySelector(
-  ".modal__input_type_description"
-);
-const newCardTitleInput = addCardForm.querySelector(".modal__input_type_title");
-const newCardUrlInput = addCardForm.querySelector(".modal__input_type_url");
+  return api.updateUserAvatar(data).catch((err) => {
+    console.error(err); // log the error to the console
+  });
+}
 
-//// Form Submit Handlers ////
 // profile form submission handler
 function handleProfileFormSubmit(data) {
-  // insert new values into the textContent property of the
-  // corresponding profile elements
-  user.setUserInfo(data);
-
-  // // close popup
-  // profilePopup.close();
-  // profilePopup.reset();
+  // change propert name of description to about
+  const updatedData = { name: data.name, about: data.description };
+
+  // Update user data on server
+  return api
+    .updateUserInfo(updatedData)
+    .then((res) => {
+      // insert new values into the textContent property of the
+      // corresponding profile elements
+      user.setUserInfo({
+        name: res.name,
+        description: res.about,
+      });
+    })
+    .catch((err) => {
+      console.error(err); // log the error to the console
+    });
 }
 
 // Add card form submission handler
 function handleAddCardFormSubmit(data) {
-  // create new card
-  const newCard = createCard(data, openImageModal);
-
-  // Add new card to begining of card gallery
-  cardsSection.addItem(newCard);
-
-  addCardPopup.close();
-  addCardPopup.reset();
-  formValidators[addCardForm.getAttribute("id")].toggleButtonState();
+  // add new card to server
+  return api
+    .addNewCard(data)
+    .then((cardData) => {
+      // create new card with the data returned from server
+      const newCard = createCard(cardData);
+
+      // Add new card to begining of card gallery
+      cardsSection.addItem(newCard);
+
+      addCardPopup.close();
+      addCardPopup.reset();
+      formValidators[addCardForm.getAttribute("id")].toggleButtonState();
+    })
+    .catch((err) => {
+      console.error(err); // log the error to the console
+    });
```

Comment:
Good job that you `catch` possible errors at the end of server requests. You can simplify it with `.catch(console.error);` because `console.error` is meant to log errors